x/ccv/provider/keeper/permissionless.go

```diff
@@ -0,0 +1,836 @@
+package keeper
+
+import (
+	"bytes"
+	errorsmod "cosmossdk.io/errors"
+	storetypes "cosmossdk.io/store/types"
+	"encoding/binary"
+	"encoding/gob"
```

Comment:
See comment [below](https://github.com/cosmos/interchain-security/pull/2117#discussion_r1726880337). Since it's for the same thing, I'm resolving this comment.